### DISCUSSION TITLE
Aligning Notification integrations (Slack, Email, OpsGene, Teams, Pager Duty, etc...)

### DISCUSSION THREAD

**Discussion by kyleaton** [great-expectations/great_expectations#5211] [Ideas]
This is a feature request from @kenparton in [our feature request Slack Channel ](https://greatexpectationstalk.slack.com/archives/CUW3WD6PK/p1653466483698499) 

> Hi all,
> Taking a look at existing great_expectations/checkpoint/actions:
> Slack
> Email
> OpsGenie
> SQS
> Pager Duty
> Microsoft Teams
> Some include notify_with (Slack, Email) while others don’t (PagerDuty, OpsGenie, Microsoft Teams, SNS). Also, some set the title of the notification in the renderer (Slack, Email, Microsoft Teams), one in the action (Page Duty), and one in util.py (OpsGenie).
> Is there any work or planned work to align these implementations?

I'm opening up this discussion here as things can get lost in Slack, and I see this as a great topic of conversation that can develop over time and hopefully recruit some contributors. Please contribute to the discussion if you have any suggestions, requests, support, or plans of action.

**Comment by KentonParton** [comment]
I'd also like to point out that the Pager Duty library `pypd` has been archived and replaced by `pdpyras` -->  https://github.com/PagerDuty/pdpyras

**Comment by KentonParton** [comment]
@kyleaton Could you shed some light on what GE's cloud offering will be using to notify customers? Based on GE's current action state, I doubt GE cloud offering will be using what's currently available.

The reason I ask is, if GE cloud is using a different notification approach, contributing makes less sense for the community as it's not going to be prioritised or improved on moving forward.

Thanks!

**Comment by KentonParton** [comment]
@kyleaton would it be possible to get some eyes on this?

**Comment by kyleaton** [comment]
Hey @KentonParton, thanks for being patient with my response. 

I would say our intention is to leverage any OSS validation action (which triggers after each checkpoint run) in the cloud offering. Any integration addition by community users is much appreciated and our cloud offering will work to include the most demanded notifications with a user interface to connect/configure them. 

Does this answer suffice?
